Description: Fix 'catch' & 'catch2' compatibility with glibc-2.34
 The gringo package contains several embedded copies of the 'catch' (CTest)
 library. All of those copies are outdated and incompatible with glibc-2.34.
 This patch updates the test sources to use the system provided versions of
 'catch' and 'catch2' instead, to fix this issue.
 C.f.: https://github.com/catchorg/Catch2/issues/2178
Author: Lukas Märdian <slyon@ubuntu.com>
Origin: vendor, Ubuntu
Bug: https://github.com/potassco/libpotassco/issues/12 (only catch1 part)
Last-Update: 2022-01-18
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: gringo/clasp/libpotassco/tests/main.cpp
===================================================================
--- gringo.orig/clasp/libpotassco/tests/main.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/libpotassco/tests/main.cpp	2025-10-02 17:38:47.215591614 +0200
@@ -16,4 +16,4 @@
 static int eh = enableDebugHeap();
 #endif
 #define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
-#include "catch.hpp"
+#include <catch/catch.hpp>
Index: gringo/clasp/libpotassco/tests/test_application.cpp
===================================================================
--- gringo.orig/clasp/libpotassco/tests/test_application.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/libpotassco/tests/test_application.cpp	2025-10-02 17:38:47.215816850 +0200
@@ -18,7 +18,7 @@
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 // FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 // IN THE SOFTWARE.
-#include "catch.hpp"
+#include <catch/catch.hpp>
 #include <potassco/application.h>
 #include <potassco/program_opts/typed_value.h>
 #include <signal.h>
Index: gringo/clasp/libpotassco/tests/test_aspif.cpp
===================================================================
--- gringo.orig/clasp/libpotassco/tests/test_aspif.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/libpotassco/tests/test_aspif.cpp	2025-10-02 17:38:47.215917047 +0200
@@ -24,7 +24,7 @@
 #ifdef _MSC_VER
 #pragma warning (disable : 4996)
 #endif
-#include "catch.hpp"
+#include <catch/catch.hpp>
 #include "test_common.h"
 #include <potassco/aspif.h>
 #include <potassco/rule_utils.h>
Index: gringo/clasp/libpotassco/tests/test_options.cpp
===================================================================
--- gringo.orig/clasp/libpotassco/tests/test_options.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/libpotassco/tests/test_options.cpp	2025-10-02 17:38:47.216015318 +0200
@@ -18,7 +18,7 @@
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 // FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 // IN THE SOFTWARE.
-#include "catch.hpp"
+#include <catch/catch.hpp>
 #include <potassco/program_opts/program_options.h>
 #include <potassco/program_opts/typed_value.h>
 #include <potassco/program_opts/errors.h>
Index: gringo/clasp/libpotassco/tests/test_smodels.cpp
===================================================================
--- gringo.orig/clasp/libpotassco/tests/test_smodels.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/libpotassco/tests/test_smodels.cpp	2025-10-02 17:38:47.216107439 +0200
@@ -22,7 +22,7 @@
 // IN THE SOFTWARE.
 //
 
-#include "catch.hpp"
+#include <catch/catch.hpp>
 #include "test_common.h"
 #include <potassco/convert.h>
 #include <sstream>
Index: gringo/clasp/libpotassco/tests/test_string_convert.cpp
===================================================================
--- gringo.orig/clasp/libpotassco/tests/test_string_convert.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/libpotassco/tests/test_string_convert.cpp	2025-10-02 17:38:47.216203130 +0200
@@ -18,7 +18,7 @@
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 // FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 // IN THE SOFTWARE.
-#include "catch.hpp"
+#include <catch/catch.hpp>
 #include <potassco/string_convert.h>
 #include <string>
 #include <sstream>
Index: gringo/clasp/libpotassco/tests/test_text.cpp
===================================================================
--- gringo.orig/clasp/libpotassco/tests/test_text.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/libpotassco/tests/test_text.cpp	2025-10-02 17:38:47.216300170 +0200
@@ -22,7 +22,7 @@
 // IN THE SOFTWARE.
 //
 
-#include "catch.hpp"
+#include <catch/catch.hpp>
 #include "test_common.h"
 #include <potassco/aspif_text.h>
 #include <potassco/aspif.h>
Index: gringo/clasp/libpotassco/tests/test_value.cpp
===================================================================
--- gringo.orig/clasp/libpotassco/tests/test_value.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/libpotassco/tests/test_value.cpp	2025-10-02 17:38:47.216395717 +0200
@@ -18,7 +18,7 @@
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 // FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 // IN THE SOFTWARE.
-#include "catch.hpp"
+#include <catch/catch.hpp>
 #include <potassco/program_opts/value_store.h>
 #include <potassco/program_opts/mapped_value.h>
 #include <potassco/program_opts/typed_value.h>
Index: gringo/clasp/tests/clause_creator_test.cpp
===================================================================
--- gringo.orig/clasp/tests/clause_creator_test.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/tests/clause_creator_test.cpp	2025-10-02 17:38:47.216484195 +0200
@@ -24,7 +24,7 @@
 #include <clasp/clause.h>
 #include <clasp/solver.h>
 #include <algorithm>
-#include "catch.hpp"
+#include <catch/catch.hpp>
 namespace Clasp { namespace Test {
 using namespace Clasp::mt;
 
Index: gringo/clasp/tests/clause_test.cpp
===================================================================
--- gringo.orig/clasp/tests/clause_test.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/tests/clause_test.cpp	2025-10-02 17:38:47.216628523 +0200
@@ -24,7 +24,7 @@
 #include <clasp/clause.h>
 #include <clasp/solver.h>
 #include <algorithm>
-#include "catch.hpp"
+#include <catch/catch.hpp>
 #ifdef _MSC_VER
 #pragma warning (disable : 4267) //  conversion from 'size_t' to unsigned int
 #pragma once
Index: gringo/clasp/tests/cli_test.cpp
===================================================================
--- gringo.orig/clasp/tests/cli_test.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/tests/cli_test.cpp	2025-10-02 17:38:47.216752431 +0200
@@ -26,7 +26,7 @@
 #include <clasp/unfounded_check.h>
 #include <clasp/logic_program.h>
 #include <fstream>
-#include "catch.hpp"
+#include <catch/catch.hpp>
 namespace Clasp {
 static bool operator==(const ScheduleStrategy& lhs, const ScheduleStrategy& rhs) {
 	return lhs.type == rhs.type && lhs.base == rhs.base && lhs.len == rhs.len && lhs.grow == rhs.grow;
Index: gringo/clasp/tests/decision_heuristic_test.cpp
===================================================================
--- gringo.orig/clasp/tests/decision_heuristic_test.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/tests/decision_heuristic_test.cpp	2025-10-02 17:38:47.216872663 +0200
@@ -27,7 +27,7 @@
 #include <clasp/clause.h>
 #include <clasp/solver.h>
 #include "lpcompare.h"
-#include "catch.hpp"
+#include <catch/catch.hpp>
 namespace Clasp { namespace Test {
 using namespace Clasp::Asp;
 /////////////////////////////////////////////////////////////////////////////////////////
Index: gringo/clasp/tests/dependency_graph_test.cpp
===================================================================
--- gringo.orig/clasp/tests/dependency_graph_test.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/tests/dependency_graph_test.cpp	2025-10-02 17:38:47.216980226 +0200
@@ -24,7 +24,7 @@
 #include <clasp/dependency_graph.h>
 #include <clasp/solver.h>
 #include "lpcompare.h"
-#include "catch.hpp"
+#include <catch/catch.hpp>
 namespace Clasp { namespace Test {
 using namespace Clasp::Asp;
 
Index: gringo/clasp/tests/dlp_builder_test.cpp
===================================================================
--- gringo.orig/clasp/tests/dlp_builder_test.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/tests/dlp_builder_test.cpp	2025-10-02 17:38:47.217082416 +0200
@@ -26,7 +26,7 @@
 #include <clasp/unfounded_check.h>
 #include "lpcompare.h"
 #include <sstream>
-#include "catch.hpp"
+#include <catch/catch.hpp>
 using namespace std;
 namespace Clasp { namespace Test {
 using namespace Clasp::Asp;
Index: gringo/clasp/tests/enumerator_test.cpp
===================================================================
--- gringo.orig/clasp/tests/enumerator_test.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/tests/enumerator_test.cpp	2025-10-02 17:38:47.217188939 +0200
@@ -28,7 +28,7 @@
 #include <clasp/model_enumerators.h>
 #include "lpcompare.h"
 #include <sstream>
-#include "catch.hpp"
+#include <catch/catch.hpp>
 using namespace std;
 namespace Clasp { namespace Test {
 using namespace Clasp::Asp;
Index: gringo/clasp/tests/facade_test.cpp
===================================================================
--- gringo.orig/clasp/tests/facade_test.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/tests/facade_test.cpp	2025-10-02 17:38:47.217392803 +0200
@@ -34,7 +34,7 @@
 #endif
 #include "lpcompare.h"
 #include <signal.h>
-#include "catch.hpp"
+#include <catch/catch.hpp>
 namespace Clasp { namespace Test {
 using namespace Clasp::mt;
 
Index: gringo/clasp/tests/literal_test.cpp
===================================================================
--- gringo.orig/clasp/tests/literal_test.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/tests/literal_test.cpp	2025-10-02 17:38:47.217592182 +0200
@@ -22,7 +22,7 @@
 // IN THE SOFTWARE.
 //
 #include <clasp/solver_types.h>
-#include "catch.hpp"
+#include <catch/catch.hpp>
 namespace Clasp { namespace Test {
 
 static bool testBin(const Literal& p) {
Index: gringo/clasp/tests/minimize_test.cpp
===================================================================
--- gringo.orig/clasp/tests/minimize_test.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/tests/minimize_test.cpp	2025-10-02 17:38:47.217716852 +0200
@@ -28,7 +28,7 @@
 #include <clasp/solver.h>
 #include <clasp/solve_algorithms.h>
 #include <algorithm>
-#include "catch.hpp"
+#include <catch/catch.hpp>
 namespace Clasp { namespace Test {
 namespace {
 struct BranchAndBoundTest {
Index: gringo/clasp/tests/parser_test.cpp
===================================================================
--- gringo.orig/clasp/tests/parser_test.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/tests/parser_test.cpp	2025-10-02 17:38:47.217882237 +0200
@@ -29,7 +29,7 @@
 #include <potassco/smodels.h>
 #include <potassco/string_convert.h>
 #include "lpcompare.h"
-#include "catch.hpp"
+#include <catch/catch.hpp>
 namespace Clasp { namespace Test {
 
 template <class Api>
Index: gringo/clasp/tests/program_builder_test.cpp
===================================================================
--- gringo.orig/clasp/tests/program_builder_test.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/tests/program_builder_test.cpp	2025-10-02 17:38:47.218098714 +0200
@@ -27,7 +27,7 @@
 #include <clasp/unfounded_check.h>
 #include <clasp/minimize_constraint.h>
 #include "lpcompare.h"
-#include "catch.hpp"
+#include <catch/catch.hpp>
 using namespace std;
 namespace Clasp { namespace Test {
 using namespace Clasp::Asp;
Index: gringo/clasp/tests/rule_test.cpp
===================================================================
--- gringo.orig/clasp/tests/rule_test.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/tests/rule_test.cpp	2025-10-02 17:38:47.218282979 +0200
@@ -27,7 +27,7 @@
 #include <clasp/solver.h>
 #include "lpcompare.h"
 #include <utility>
-#include "catch.hpp"
+#include <catch/catch.hpp>
 namespace Potassco {
 static bool operator==(const LitSpan& lhs, const LitSpan& rhs) {
 	return lhs.size == rhs.size && std::equal(begin(lhs), end(lhs), begin(rhs));
Index: gringo/clasp/tests/satelite_test.cpp
===================================================================
--- gringo.orig/clasp/tests/satelite_test.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/tests/satelite_test.cpp	2025-10-02 17:38:47.218405045 +0200
@@ -28,7 +28,7 @@
 #include <clasp/parser.h>
 #include <algorithm>
 #include <sstream>
-#include "catch.hpp"
+#include <catch/catch.hpp>
 #ifdef _MSC_VER
 #pragma warning (disable : 4267) //  conversion from 'size_t' to unsigned int
 #pragma once
Index: gringo/clasp/tests/solver_test.cpp
===================================================================
--- gringo.orig/clasp/tests/solver_test.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/tests/solver_test.cpp	2025-10-02 17:39:10.987343206 +0200
@@ -28,7 +28,7 @@
 #if CLASP_HAS_THREADS
 #include <clasp/mt/parallel_solve.h>
 #endif
-#include "catch.hpp"
+#include "catch/catch.hpp"
 
 
 namespace Clasp { namespace Test {
Index: gringo/clasp/tests/test_main.cpp
===================================================================
--- gringo.orig/clasp/tests/test_main.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/tests/test_main.cpp	2025-10-02 17:38:47.218942259 +0200
@@ -22,7 +22,7 @@
 // IN THE SOFTWARE.
 //
 #define CATCH_CONFIG_RUNNER
-#include "catch.hpp"
+#include <catch/catch.hpp>
 #if defined (_MSC_VER) && _MSC_VER >= 1200
 #include <crtdbg.h>
 #endif
Index: gringo/clasp/tests/unfounded_check_test.cpp
===================================================================
--- gringo.orig/clasp/tests/unfounded_check_test.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/tests/unfounded_check_test.cpp	2025-10-02 17:38:47.219058339 +0200
@@ -26,7 +26,7 @@
 #include <clasp/clause.h>
 #include "lpcompare.h"
 #include <memory>
-#include "catch.hpp"
+#include <catch/catch.hpp>
 namespace Clasp { namespace Test {
 using namespace Clasp::Asp;
 
Index: gringo/clasp/tests/weight_constraint_test.cpp
===================================================================
--- gringo.orig/clasp/tests/weight_constraint_test.cpp	2025-10-02 17:38:47.219230736 +0200
+++ gringo/clasp/tests/weight_constraint_test.cpp	2025-10-02 17:38:47.219191011 +0200
@@ -24,7 +24,7 @@
 #include <clasp/weight_constraint.h>
 #include <clasp/solver.h>
 #include <algorithm>
-#include "catch.hpp"
+#include <catch/catch.hpp>
 using namespace std;
 
 namespace Clasp { namespace Test {
